diksha/diksha.init.js

```diff
@@ -52,11 +52,22 @@ let initializeDikshaData = (path) => {
 
 let processEcarFiles = (filePath) => {
     return readdir(filePath)
-        .then(files => {	
+        .then(files => {
             return files
                 .filter(file => file.endsWith('.ecar'))
                 .reduce((chainedExtractPromises, file) => {
-                    return chainedExtractPromises.then(() => extractFile(filePath, file));
+                    return chainedExtractPromises.then(
+                        () => {
+                            console.log(`Extraction of ${file} completed.`);
+
+                            return extractFile(filePath, file);
+                        },
+                        err => {
+                            console.log(`Extraction of ${file} failed.`);
+                            console.log(err);
+
+                            return extractFile(filePath, file);
```

Comment:
makes no sense, this operation gave an error earlier, so there's a very high chance that it will give again, and there's no one to catch it :( ... it hurts 
@rak16

Reply:
if you want to make sure this gets done anyway, make a function out of it, call it recursively...but that's a bad idea, may end up in infinite loop

Reply:
accept errors, handle gracefully, provide proper feedback and exit gracefully or continue :smile: :+1:

Reply:
@rishabh1403 There is a `catch` at the place where `processEcarFiles` is being called so that is covered.